IndoorMapsFrontend/src/components/forms/AutoCompleteResults.tsx

```diff
@@ -41,7 +47,24 @@ const AutoCompleteResults = ({ getGeocoder, searchString, chooseAutocompleteResu
     }, [debouncedValue])
 
     const listOfAutocompleteElements = data.getAutocomplete.items.map(item => {
-        return (<li className="autocompleteResultItem" key={item.id}><Button onClick={() => chooseAutocompleteResult(item)}>{item.title}</Button></li>)
+        let titleToDispaly = item.title;
+        if(item.highlights && item.highlights.title){
+            let newTitle = "";
+            // supposedly the newer versions of ts are better at using filters to remove null and undefired, here we need as number[]
+            const starts: number[] = item.highlights.title.map((titleHighlight) => titleHighlight.start).filter((val) => (val !== null && val !== undefined)) as number[];
+            const ends: number[] = item.highlights.title.map((titleHighlight) => titleHighlight.end).filter((val) => (val !== null && val !== undefined)) as number[];
+            for(let i = 0; i < titleToDispaly.length; i++){
```

Comment:
You can add a comment here to explain what is going on

Reply:
Change Applied: https://github.com/Theo-s-Meta-Capstone/IndoorMaps/commit/e3555e906d4cc8abe53f7a105c8e1f0556a13b89